blocks/extensions/training.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+from blocks.extensions import TrainingExtension
+
+
+class SharedVariableModifier(TrainingExtension):
```

Comment:
You should probably inherit from `SimpleExtension` instead and move `after_batch` to a `do` method. The advantage is then that the user could choose to e.g. run this only once an epoch, or every `n` batches, etc. making it a bit more flexible (and fixing the current Travis error).